dev-client/src/components/dataRequirements/RestrictByRequirements.tsx

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright © 2024 Technology Matters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import {useEffect} from 'react';
+
+type Requirement = {
+  data: any;
+  doIfMissing?: () => void;
+};
+
+const dataExists = (data: any) => {
```

Comment:
nit: i think you could use `object | undefined | null` to be a bit less permissive here

Reply:
This may change when we add permissions stuff, but happy to at least start out more permissive and adjust later if needed 👍